src/services/user/hooks/handle-facebook-create.js

```diff
@@ -0,0 +1,52 @@
+
+const handleFacebookCreate = () => (hook) => {
+  // Pass through this hook if there's no facebook data
+  if (!hook.data.facebookId
+    || !hook.data.facebook
+    || !hook.data.facebook.email
+    || (!hook.data.facebook.name
+      && !hook.data.facebook.first_name
+      && !hook.data.facebook.last_name)
+    ) {
+    return hook;
```

Comment:
Right. (Actually, I still don't understand how different. Their doc said we should return Promise but I see some example code just return hook while some return Promise.)